lib/runtime/wazero/imports.go

```diff
@@ -781,33 +782,35 @@ func ext_crypto_finish_batch_verify_version_1(ctx context.Context, m api.Module)
 }
 
 func ext_trie_blake2_256_root_version_1(ctx context.Context, m api.Module, dataSpan uint64) uint32 {
+	return ext_trie_blake2_256_root_version_2(ctx, m, dataSpan, 0)
+}
+
+func ext_trie_blake2_256_root_version_2(ctx context.Context, m api.Module, dataSpan uint64, version uint32) uint32 {
 	rtCtx := ctx.Value(runtimeContextKey).(*runtime.Context)
 	if rtCtx == nil {
 		panic("nil runtime context")
 	}
 
-	data := read(m, dataSpan)
-
-	t := trie.NewEmptyTrie()
-
-	type kv struct {
-		Key, Value []byte
+	stateVersionBytes, _ := m.Memory().Read(version, 4)
+	stateVersion, err := trie.ParseVersion(binary.LittleEndian.Uint32(stateVersionBytes))
+	if err != nil {
+		logger.Errorf("failed parsing state version: %s", err)
+		return 0
```

Comment:
You don't need to read the version from memory, the state version is already given to you through the `version` argument, the version is either `0` or `1`.

https://spec.polkadot.network/chap-host-api#id-version-2-prototype-9

So you can just:
```go
stateVersion, err := trie.ParseVersion(version)
```